code/modules/projectiles/guns/flamer.dm

```diff
@@ -655,7 +531,11 @@ GLOBAL_LIST_EMPTY(flamer_particles)
 	flame_max_wall_pen_wide = 1
 
 /obj/item/weapon/gun/flamer/hydro_cannon/flame_turf(turf/turf_to_ignite, mob/living/user, burn_time, burn_level, fire_color = "red", direction = NORTH)
-	var/obj/flamer_fire/current_fire = locate(/obj/flamer_fire) in turf_to_ignite
+	var/obj/fire_base/fire/current_fire = locate(/obj/fire_base/fire) in turf_to_ignite
+	if(current_fire)
+		qdel(current_fire)
+	current_fire = null
+	current_fire = locate(/obj/fire_base/melting_fire) in turf_to_ignite
```

Comment:
locate(/obj/fire)

in future if you dont understand what I asked of you please ask instead of pointlessly repathing it like this